src/js/harness.js

```diff
@@ -292,25 +292,34 @@ gpii.test.couchdb.harness.clearInterval = function (that) {
  */
 gpii.test.couchdb.harness.startMonitoring = function (that) {
     if (that.options.monitorContainer) {
-        that.monitorInterval = setInterval(
-            function () {
-                var isUpPromise = that.worker.isUp();
-                isUpPromise.then(
-                    function (isUp) {
-                        fluid.log(fluid.logLevel.TRACE, "Container " + isUp ? "is" : "is not" + " up.");
-
-                        if (!isUp) {
-                            that.events.onCouchMissing.fire();
-                        }
-                    },
-                    fluid.fail
-                );
-            },
+        that.monitorTimeout = setTimeout(
+            that.monitorContainerOnce,
             that.options.containerMonitoringInterval
         );
     }
 };
 
+gpii.test.couchdb.harness.monitorContainerOnce = function (that) {
```

Comment:
In discussions with @amb26, we discussed improving this, his (lightly edited) remarks below:

> The logic of this set of functions, collectively, is pretty obscure.  I recommend making a "timer" subcomponent that embodies the timer, which represents clearing the timer by its own destruction.  There are a few patterns for this in  gpii-app:
> 
> https://github.com/GPII/gpii-app/blob/master/src/shared/utils.js#L35